### PR TITLE
lnav 0.12.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
only test failure on 12-arm

```
  ==> /opt/homebrew/Cellar/lnav/0.12.1/bin/lnav -V
  Error: lnav: failed
  An exception occurred within a child process:
    BuildError: Failed executing: /opt/homebrew/Cellar/lnav/0.12.1/bin/lnav -V
```

**Comment by carlocab** [comment]
Upstream have a crash report at https://github.com/tstack/lnav/issues/1245. No idea if related, but let's see if their fixes for it help...

**Comment by tstack** [comment]
> No idea if related, but let's see if their fixes for it help...

The crash in that issue happens when displaying the UI, but `lnav -V` would only print out the version.  So, I highly doubt they're related.

The formula change currently has the patches being applied in the wrong order, you need to switch them around.

**Comment by tstack** [comment]
The following change fixes the `lnav -V` crash on macos-12-arm64:

https://github.com/tstack/lnav/commit/884e6b7415c562621487abe922fd3042da71ff32

**Comment by chenrui333** [comment]
> The following change fixes the `lnav -V` crash on macos-12-arm64:
> 
> [tstack/lnav@884e6b7](https://github.com/tstack/lnav/commit/884e6b7415c562621487abe922fd3042da71ff32)

great, thanks for the pointer, adopting the patch now.

File: Formula/l/lnav.rb [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
   homepage "https://lnav.org/"
-  url "https://github.com/tstack/lnav/releases/download/v0.12.0/lnav-0.12.0.tar.gz"
-  sha256 "a6f51a5019df1870c6a5889e337cf2cfc1365b04d81a0602f7f071ec971f5997"
   license "BSD-2-Clause"
 
+  stable do
+    url "https://github.com/tstack/lnav/releases/download/v0.12.1/lnav-0.12.1.tar.gz"
+    sha256 "d4565fbe29ad00e1e00efc6bff68e2068e102ace95ff296610d2bd6c04b13d67"
+
+    # patch to fix static init issue
```
Comment (MikeMcQuaid) [inline Formula/l/lnav.rb:10]: Can you link to the issue or PR?
```suggestion
    # patch to fix static init issue
    # https://github.com/???
```

Comment (chenrui333) [reply Formula/l/lnav.rb:10]: @MikeMcQuaid this is a direct commit patch from @tstack, https://github.com/tstack/lnav/commit/884e6b7415c562621487abe922fd3042da71ff32

see discussions above.

Comment (MikeMcQuaid) [reply Formula/l/lnav.rb:10]: @chenrui333 That's fine. I just don't know what "static init issue" means and why we want/need this patch, thanks!

**Review by MikeMcQuaid — COMMENTED** [review]
patch needs better comment

File: Formula/l/lnav.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/l/lnav.rb:10]: oh, it caused some crash when starting the app. (I can maybe improve the comment a bit)

Comment (tstack) [reply Formula/l/lnav.rb:10]: It was an issue with C++ static initializer ordering.  A variable was being used before it was initialized and caused a crash.

File: Formula/l/lnav.rb [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+  stable do
+    url "https://github.com/tstack/lnav/releases/download/v0.12.1/lnav-0.12.1.tar.gz"
+    sha256 "d4565fbe29ad00e1e00efc6bff68e2068e102ace95ff296610d2bd6c04b13d67"
+
+    # patch to fix static init issue running on 12-arm
+    # upstream release request, https://github.com/tstack/lnav/issues/1250
+    patch do
+      url "https://github.com/tstack/lnav/commit/884e6b7415c562621487abe922fd3042da71ff32.patch?full_index=1"
+      sha256 "a6e6bf235159f3a4a3af82a01099b4aaf70e102f84e925d8d2f8d867b0d7f2b6"
+    end
```
Comment (MikeMcQuaid) [inline Formula/l/lnav.rb:15]: Is it worth applying patch only on 12-arm conditional on Clang versions/hardware/etc.?

Comment (tstack) [reply Formula/l/lnav.rb:15]: No, it could happen anywhere,  theoretically.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @chenrui333 and @tstack!

**Pull request merged**